WebGoatCore/Models/CreditCard.cs

```diff
@@ -193,13 +203,9 @@ private bool CardExistsForUser()
             var document = ReadCreditCardFile();
             try
             {
-                var xElement = document.Descendants("CreditCard").FirstOrDefault(c => c.Element("Username").Value.Equals(Username));
-                if (!xElement.HasElements)
-                {
-                    throw new NullReferenceException(string.Format("No card found for {0}.", Username));
-                }
+                GetCreditCardXmlElement(document);
```

Comment:
Maybe GetCreditCardXmlElement could return null if it could not find the element?

Reply:
How will it help us? We still need to report some kind of issue, because this situation is not expected. I don't see a viable solution regarding what to do with that null value later (assuming we do not want to silently ignore it).